code/go/0chain.net/smartcontract/minersc/fees.go

```diff
@@ -506,7 +524,15 @@ func (msc *MinerSmartContract) payFees(t *transaction.Transaction,
 	if len(shardersIDs) > 0 {
 		seed := b.GetRoundRandomSeed()
 		randS := rand.New(rand.NewSource(seed))
-		mbShardersIDs := getRegisterShardersInMagicBlock(balances, shardersIDs)
+		mbShardersIDs, err := getRegisterShardersInMagicBlock(balances, gn, shardersIDs)
+		if err != nil {
```

Comment:
this also will break MPT consistency if network is partially updated

Reply:
@dabasov the code won't apply when VC is switched off, that's why I asked to update all nodes to latest image before enabling the VC. We can add hardfork to make it compatible, but that could make the code complex as we have so many VC changes. Actualy, we should not enable the VC when the network is partially updated.